### ISSUE TITLE
[BUG] Restoring selected values occurs error

### ISSUE THREAD

**Issue by qstiegler** [bithost-gmbh/ngx-mat-select-search#376]
In mode multi if you have selected one and then search for another value where the selected option from before is not part of the result, it deselects the previous selected one and only the current one is selected. This happens because a JS error occurs:

```
ERROR TypeError: Cannot add property 1, object is not extensible
```

The error occurs in the following line of code: https://github.com/bithost-gmbh/ngx-mat-select-search/blob/782e635ce57add4696bbccc89b3b1f68e2f9f3c5/src/app/mat-select-search/mat-select-search.component.ts#L618

This happens because you're pushing a value to the readonly variable `values`. You need to create a new instance of the array and use this instead.

**Comment by macjohnny** [comment]
released in `4.1.2`

**Comment by angelaki** [comment]
Hey, this fix just broke my behavior! By modifying a copy of values but not the original one the `(selectionChange)` event of the control does not contain all the elements but just the newly selected.

@qstiegler Can you show me your usecase where version 4.1.1 threw an exception for you so I can maybe fix this?

**Comment by KristofGilis** [comment]
I created a PR #467 to fix this issue.

The array just must be recreated instead of pushing the items inside the existing array, that could be not extensible.

**Comment by macjohnny** [comment]
thanks @KristofGilis!
This was released as part of [v7.0.8](https://github.com/bithost-gmbh/ngx-mat-select-search/releases/tag/7.0.8)
